src/content/plugins/chunks-webpack-plugin.md

```diff
@@ -0,0 +1,322 @@
+    ).join('');
+
+    // Generate all HTML style tags with CDN prefix and defer attribute
+    const scripts = chunksSorted.scripts.map(chunkJs =>
+      `<script defer src="https://cdn.domain.com/${chunkJs}"></script>`
+    ).join('');
+
+    return { styles, scripts };
+  }
+});
+```
+
+T> The arrow function syntax allow you to access the class properties.
+
+W> The function provides more flexibility by replacing the default behavior. Follow the example above to make sure it works.
+
+The function must return an object with the following format:
+
+```json
+{
+  "styles": "",
+  "scripts": ""
+}
+```
+
+The `customFormatTags` function has two parameters:
+
+#### `chunksSorted`
+
+`object`
+
+The list of the chunks sorted by type: `styles` and `scripts`.
+
+#### `Entrypoint`
+
+`object`
+
+The object is included in every single ChunkGroup. The variable contains all information about the current entry point; log it on the console for more details.
+
+W> Use this variable only for a full customization if the `chunksSorted` variable does not meet your needs.
+
+### `generateChunksManifest`
+
+`boolean = false`
+
+Tells the plugin whether to generate the `chunks-manifest.json`. The file contains the list of all chunks grouped by entry points.
+
+```js
+new ChunksWebpackPlugin({
+  generateChunksManifest: true
+});
+```
+
+Example of the output of the `chunks-manifest.json` file:
+
+```json
+{
+  "app-a": {
+    "styles": ["/dist/css/vendors~app-a~app-b.css", "/dist/css/app-a.css"],
+    "scripts": ["/dist/js/vendors~app-a~app-b.js", "/dist/js/app-a.js"]
+  },
+  "app-b": {
+    "styles": ["/dist/css/vendors~app-a~app-b.css", "/dist/css/app-b.css"],
+    "scripts": ["/dist/js/vendors~app-a~app-b.js", "/dist/js/app-b.js"]
+  }
+}
+```
+
+### `generateChunksFiles`
+
+`boolean = true`
+
+Tells the plugin whether to generate the HTML files.
+
+```js
+new ChunksWebpackPlugin({
+  generateChunksFiles: false
+});
+```
+
+W> When set to `false`, HTML files will not be generated. It can only be useful together with  `generateChunksManifest` option set to `true` for custom generation of the HTML files.
+
+### Caching
+
+The [webpack caching](/guides/caching/) feature allows you to generate HTML files that include hash in the filename.
+
+```js
+const ChunksWebpackPlugin = require('chunks-webpack-plugin');
+const path = require('path');
+
+module.exports = {
+  entry: 'main.js',
+  output: {
+    path: path.resolve(__dirname, './dist'),
+    filename: 'bundle.[contenthash].js'
+  },
+  plugins: [new ChunksWebpackPlugin()]
+};
+```
+
+This will generate the following HTML files with hash in the filename.
+
+__main-styles.html__
+
+```html
+<link rel="stylesheet" href="main.72dd90acdd3d4a4a1fd4.css" />
+```
+
+__main-scripts.html__
+
+```html
+<script src="main.72dd90acdd3d4a4a1fd4.js"></script>
+```
+
+### Multiple entry points
+
+Example of the webpack configuration with multiple entry points which share common code:
```

Comment:
@EugeneHlushko I think this example can be ambiguous because I've write `common code` but there is no optimization in the webpack configuration. What do you think?

We can update the text to clarify this point

```text
Example of the webpack configuration with multiple entry points which share common code with the `splitChunks` parameter.
```

And add the following lines in the config like the [example](https://github.com/yoriiis/chunks-webpack-plugin/blob/master/example/webpack.config.js#L111-L114) ?

```js
optimization: {
	splitChunks: {
		chunks: 'all'
	}
}
```

Reply:
makes sense

Reply:
Update in this [commit](https://github.com/webpack/webpack.js.org/pull/3771/commits/e0021d249159e494ba617348eb93bb1d37b35530).